### ISSUE TITLE
Tenacity 7.0 breaks test suite

### ISSUE THREAD

**Issue by peterschutt** [MikeWooster/api-client#61]
[Tenacity has removed `tenacity.compat.make_retry_state`](https://github.com/jd/tenacity/pull/274) which features heavily in `test_retrying.py`.

https://github.com/MikeWooster/api-client/blob/5a1370feda81e95254b14e8a1ae5642f3209d595/tests/test_retrying.py#L6

**To Reproduce**
1. Create new virtualenv and activate.
2. `python -m pip install .[test]`
3. `pytest`

**Environment**
```
(.venv) peter@petes-laptop:~/peterschutt/api-client$ python -m pip freeze
api-client @ file:///home/peter/peterschutt/api-client
attrs==20.3.0
certifi==2020.12.5
chardet==4.0.0
coverage==5.5
idna==2.10
iniconfig==1.1.1
multidict==5.1.0
packaging==20.9
pluggy==0.13.1
py==1.10.0
pyparsing==2.4.7
pytest==6.2.2
pytest-cov==2.11.1
pytest-env==0.6.2
PyYAML==5.4.1
requests==2.25.1
requests-mock==1.8.0
six==1.15.0
tenacity==7.0.0
toml==0.10.2
urllib3==1.26.4
vcrpy==4.1.1
wrapt==1.12.1
yarl==1.6.3
```
I've run out of time to look at a fix right now, but will do so at my next opportunity if you haven't already resolved.

**Comment by peterschutt** [comment]
Prob need to include `tenacity70` in tox envlist.

**Comment by MikeWooster** [comment]
fixed in #64
